sys/net/application_layer/gcoap/fileserver.c

```diff
@@ -264,13 +453,65 @@ static ssize_t gcoap_fileserver_directory_handler(coap_pkt_t *pdu, uint8_t *buf,
     return (uintptr_t)buf - (uintptr_t)pdu->hdr;
 }
 
+static ssize_t _put_directory(coap_pkt_t *pdu, uint8_t *buf, size_t len,
```

Comment:
Should creating the directory be a separate request or should it happen automatically when a file is PUT into a directory that does not exist yet?

Reply:
According to the comments in the doc, the directory should be created implicitly. But I think it is simpler if we assume that the directory must be created in advance because if something goes wrong during the file creation, it would be more effort for us to revert the created directories.

Reply:
I wrote that under the influence of git (helluva drug ;-) ) -- but I've come to think that explicit creation is cleaner (also because it's symmetric with that we don't delete directories when deleting the last files in them); that it's easier to to implement seems to confirm that that's the better direction.

coap-kitchensink-01 will recommend that an empty PUT (with no content-format) be used to create a directory. (It'd be an error to have a payload in that.)